src/Test.UnitTests.Sarif/Baseline2/V2ResultMatcherTests.cs

```diff
@@ -161,6 +186,47 @@ public void V2ResultMatcher_TwoAdded()
             matches.Where(m => m.PreviousResult == null || m.CurrentResult == null).Should().HaveCount(2);
         }
 
+        [Fact]
+        public void V2ResultMatcher_TwoRulesSameLine()
+        {
+            // Verify that two Results with identical locations but different RuleIDs sort in a consistent order
+            // so that they will be matched with each other.
+
+            // COMPLEX: Getting to the "sufficiently similar" check requires that they don't match beforehand due to:
+            //  - Identical where, so the results must move
+            //  - Unique identical what, so there must be other unmatched results which share every trait with the ones we're checking
+
+            Run firstRun = SampleRun.DeepClone();
+
+            // Copy the first Result and change the Rule only (they'll have same Message, Fingerprints, Location)
+            firstRun.Results[1] = firstRun.Results[0].DeepClone();
+            firstRun.Results[1].RuleId = "NewRuleId";
+
+            // Make another copy of each result and move them so that the results won't have any per-rule unique traits
+            firstRun.Results.Add(firstRun.Results[0].DeepClone());
+            firstRun.Results.Add(firstRun.Results[1].DeepClone());
+            firstRun.Results[5].Locations[0].PhysicalLocation.Region.StartLine += 1;
```

Comment:
>5 [](start = 29, length = 1)

Consider counting the results before adding the new ones so you can calculate these indices instead of hard-coding them (resilient if the test file ever sprouts new results).

Reply:
Good idea.